components/server/src/test/java/com/hotels/styx/server/netty/codec/UrlDecoderTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.server.netty.codec;
+
+import com.hotels.styx.api.Url;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.netty.handler.codec.http.HttpHeaders.Names.HOST;
+import static io.netty.handler.codec.http.HttpMethod.GET;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class UrlDecoderTest {
+    @Test
+    public void decodesOriginForm() {
```

Comment:
This test is redundant

Reply:
I have made some changes to the test files. Please have another look.